src/frontend/src/lib/services/rest.services.ts

```diff
@@ -0,0 +1,37 @@
+export interface RestRequestParams<R, E = unknown, S = void> {
+	request: () => Promise<R>;
+	onSuccess: (response: R) => S | undefined;
+	onError?: (error: E) => S | undefined;
+	onRetry?: (options: { error: E; retryCount: number }) => Promise<void>;
+	maxRetries?: number;
+}
+
+/**
+ *
+ */
+export const restRequest = async <R, E, S>({
+	request,
+	onSuccess,
+	onError,
+	onRetry,
+	maxRetries = 10
```

Comment:
Do we want 10 as default? I imagined something smallerin a first time, like 3

Reply:
ah right! yes, totally random eheh